crates/runtime/src/dataconnector/spiceai.rs

```diff
@@ -159,13 +161,18 @@ impl DataPublisher for SpiceAI {
         let dataset_path = Self::spice_dataset_path(dataset);
         let mut client = self.flight.client.clone();
         let spaced_trace = Arc::clone(&self.spaced_trace);
+        let api_key = self.api_key.clone();
         Box::pin(async move {
             tracing::debug!("Adding data to {dataset_path}");
 
-            client
-                .publish(&dataset_path, data_update.data)
-                .await
-                .context(UnableToPublishDataSnafu)?;
+            if let Err(e) = client.publish(&dataset_path, data_update.data).await {
+                if let flight_client::Error::PermissionDenied {} = e {
+                    tracing::error!("Permission denied when publishing data to Spice AI at '{dataset_path}' - does the API Key ('{api_key}') match the app that owns the dataset?");
```

Comment:
It's not safe to print the API key in the logs, right?

Reply:
The increased observability outweighs the downside here. Otherwise people might think "I do have permissions to this dataset" without realizing that the API key the runtime is configured to use is the thing that needs to match, which leads to the confusion in the linked bug.